### PR TITLE
Bug Fix in Orca Pytorch Quickstart

### PR DESCRIPTION
**Pull request by GZHoffie**
Fix some bugs as discussed in #3632. Also fixed some outdated hyperlinks (e.g. link to `Orca XShards`) in markdown and notebook.

### CODE REVIEW

**Comment by yangw1234** [comment]
you may want to "git rebase" on master to get a clean list of commits.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ def optim_creator(model, config):
 
 ### **Step 3: Define Train Dataset**
 
-You can define the dataset using a *Data Creator Function* that returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](./data).
+You can define the dataset using a *Data Creator Function* that returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/Orca/data.md).
```
Comment (jason-dai) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md:82]: use relative path, not absolute path

File: docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "id": "view-in-github"
    },
    "source": [
-    "<a href=\"https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
+    "![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb)"
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb:10]: the image link is wrong?

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 3]
```diff
@@ -126,13 +139,12 @@
...
     "!conda update --channel defaults --all --yes\n",
     "\n",
     "# Append to the sys.path\n",
-    "import sys\n",
     "_ = (sys.path\n",
-    "        .append(\"/usr/local/lib/python3.6/site-packages\"))\n",
+    "        .append(f\"/usr/local/lib/python{version_info.major}.{version_info.minor}/site-packages\"))\n",
     "\n",
     "os.environ['PYTHONHOME']=\"/usr/local\""
    ]
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:150]: in below, change 

```
    **Distributed PyTorch (BigDL Backend) using Orca APIs**\n",
    "\n",
    "In this guide we will describe how to scale out PyTorch (v1.5+) programs with BigDL backend using Orca in 4 simple steps."
```

to

    "## **Distributed PyTorch using Orca APIs**\n",
    "\n",
    "In this guide we will describe how to scale out PyTorch (v1.5+) programs using Orca in 4 simple steps."

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 4]
```diff
@@ -303,7 +315,17 @@
    "source": [
     "### **Step 3: Define Train Dataset**\n",
     "\n",
-    "You can define the dataset using standard [Pytorch DataLoader](https://pytorch.org/docs/stable/data.html)."
+    "You can define the dataset using standard [Pytorch DataLoader](https://pytorch.org/docs/stable/data.html). We may also use a [data creator function](./pytorch_lenet_mnist_data_creator_func.ipynb) or [Orca SparkXShards](https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/Orca/data.md). The data creator function should have no parameters."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:318]: No need to mention data creator function or XShards here

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb [hunk 5]
```diff
@@ -7,7 +7,7 @@
     "id": "view-in-github"
    },
    "source": [
-    "<a href=\"https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
+    "![](../../../../readthedocs/image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb) "
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb:10]: the image and url are no longer in the same line?

File: docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb [hunk 6]
```diff
@@ -126,13 +139,12 @@
...
     "!conda update --channel defaults --all --yes\n",
     "\n",
     "# Append to the sys.path\n",
-    "import sys\n",
     "_ = (sys.path\n",
-    "        .append(\"/usr/local/lib/python3.6/site-packages\"))\n",
+    "        .append(f\"/usr/local/lib/python{version_info.major}.{version_info.minor}/site-packages\"))\n",
     "\n",
     "os.environ['PYTHONHOME']=\"/usr/local\""
    ]
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb:150]: change below
```
    "## **Distributed PyTorch (Ray Backend) using Orca APIs**\n",
    "\n",
    "In this guide we will describe how to scale out PyTorch (v1.5+) programs with `pytorch_distributed` backend using Orca in 4 simple steps."
```
to
```
    "## **Distributed PyTorch using Orca APIs**\n",
    "\n",
    "In this guide we will describe how to scale out PyTorch programs using `torch.distributed` package on Orca in 4 simple steps."
```

File: docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb [hunk 7]
```diff
@@ -314,7 +326,7 @@
    "source": [
     "### **Step 3: Define Train Dataset**\n",
     "\n",
-    "You can define the dataset using a *Data Creator Function* that has two parameters `config` and `batch_size` and returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](./data)."
+    "You can define the dataset using a *Data Creator Function* that has two parameters `config` and `batch_size` and returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](../../../Orca/data.md)."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb:329]: no need to mention XShards here.

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb [hunk 8]
```diff
@@ -0,0 +1,492 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "view-in-github"
+   },
+   "source": [
+    "![](../../../../readthedocs/image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb)"
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb:10]: the image and url are not in the same line in the rendered HTML: https://github.com/intel-analytics/analytics-zoo/blob/67495743966f83409951681a6ee30000850410ad/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb [hunk 9]
```diff
@@ -0,0 +1,492 @@
...
+    "!pip install jep==3.9.0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ATJ4YPAS2TQm"
+   },
+   "source": [
+    "## **Distributed PyTorch (BigDL Backend) using Orca APIs**\n",
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb:194]: Change to Distributed PyTorch using Orca APIs

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb [hunk 10]
```diff
@@ -0,0 +1,492 @@
...
+    "\n",
+    "In this guide we will describe how to scale out PyTorch (v1.5+) programs with BigDL backend using Orca in 4 simple steps."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb:196]: In this guide we will describe how to scale out PyTorch programs using Orca in 4 simple steps.

File: docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb [hunk 11]
```diff
@@ -0,0 +1,492 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "o-SeParPNtK8"
+   },
+   "source": [
+    "### **Step 3: Define Train Dataset**\n",
+    "\n",
+    "You can define the dataset using standard [Pytorch DataLoader](https://pytorch.org/docs/stable/data.html) (see example [here](./pytorch_lenet_mnist.ipynb)). We may also use a data creator function (as shown below) or [Orca SparkXShards](https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/Orca/data.md). The data creator function should have no parameters."
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist_data_creator_func.ipynb:318]: You may use a Data Creator Function for your input data  (as shown below), especially when the data size is very large.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md (hunk 1 above)
Comment (GZHoffie) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md]: While rendering readthedocs, I found that the relative path is not working because the `data.md` is not in the directory of `readthedocs/`. Will changing it to `Orca/Overview/data-parallel-processing` (which also has a paragraph that talks about `XShards`) work?

Comment (GZHoffie) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md]: Actually there are multiple hyperlinks in the current readthedocs that describe the `XShards` API and link to `./data`, which doesn't work anymore.

**Comment by GZHoffie** [comment]
Jenkins passed for the colab notebooks: http://10.239.47.210:18888/job/ZOO-PR-Quickstart-notebook/225/.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md (hunk 1 above)
Comment (jason-dai) [reply docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md]: > Actually there are multiple hyperlinks in the current readthedocs that describe the `XShards` API and link to `./data`, which doesn't work anymore.

Should be this file: https://github.com/intel-analytics/analytics-zoo/blob/master/docs/readthedocs/source/doc/Orca/Overview/data-parallel-processing.md

File: docs/docs/colab-notebook/orca/quickstart/keras_lenet_mnist.ipynb [hunk 12]
```diff
@@ -1,377 +1,390 @@
 {
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/keras_lenet_mnist.ipynb:1]: why is this file changed?

Comment (GZHoffie) [reply docs/docs/colab-notebook/orca/quickstart/keras_lenet_mnist.ipynb:1]: I don't know why... I only opened this file using jupyter notebook, copied a line, and closed it.

Comment (yangw1234) [reply docs/docs/colab-notebook/orca/quickstart/keras_lenet_mnist.ipynb:1]: If you do not meant to change this file, you should not "git add" it to your commit.

Comment (GZHoffie) [reply docs/docs/colab-notebook/orca/quickstart/keras_lenet_mnist.ipynb]: Changes are now reverted, sorry.

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md [hunk 13]
```diff
@@ -79,7 +79,7 @@ def optim_creator(model, config):
 
 ### **Step 3: Define Train Dataset**
 
-You can define the dataset using a *Data Creator Function* that returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](./data).
+You can define the dataset using a *Data Creator Function* that returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](../../../../../docs/Orca/data.md).
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-pytorch-distributed-quickstart.md:82]: this link is not consistent with the one in `orca-pytorch-quickstart.md`

File: docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb [hunk 14]
```diff
@@ -314,7 +326,7 @@
    "source": [
     "### **Step 3: Define Train Dataset**\n",
     "\n",
-    "You can define the dataset using a *Data Creator Function* that has two parameters `config` and `batch_size` and returns a PyTorch `DataLoader`. Orca also supports [Orca SparkXShards](./data)."
+    "You can define the dataset using a *Data Creator Function* that has two parameters `config` and `batch_size` and returns a PyTorch `DataLoader`."
    ]
   },
   {
```
Comment (jason-dai) [inline docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb:332]: in below, change `First, Create an Estimator.` to `First, create an Estimator and set its backend to "torch_distributed".`

**Comment by GZHoffie** [comment]
Do I need to add a link in the quickstart markdown to `pytorch_distributed_lenet_mnist.ipynb`? Currently, the notebook is not reachable from readthedocs.

**Comment by jason-dai** [comment]
> Do I need to add a link in the quickstart markdown to `pytorch_distributed_lenet_mnist.ipynb`? Currently, the notebook is not reachable from readthedocs.

You may add links to https://analytics-zoo.readthedocs.io/en/latest/doc/UserGuide/notebooks.html first; we'll add other links later.

**Comment by GZHoffie** [comment]
Thanks, all the comments above are now resolved (the backend comment is adjusted in commit `Adjust detail in notebook` but is somehow not displayed).

File: docs/readthedocs/source/doc/UserGuide/notebooks.md [hunk 15]
```diff
@@ -20,3 +20,7 @@
 
 ![](../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb) &nbsp;![](../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_lenet_mnist.ipynb)
 
+- **PyTorch (`torch_distributed` backend) Quickstart**
+
+![](../../../../image/colab_logo_32px.png)[Run in Google Colab](https://colab.research.google.com/github/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb) &nbsp;![](../../../../image/GitHub-Mark-32px.png)[View source on GitHub](https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/colab-notebook/orca/quickstart/pytorch_distributed_lenet_mnist.ipynb)i
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/UserGuide/notebooks.md]: Is the "i" in the end intended?

**Comment by yangw1234** [comment]
LGTM, let's merge it after jenkins passed

**Comment by GZHoffie** [comment]
Jenkins passed: http://10.239.47.210:18888/job/ZOO-PR-Quickstart-notebook/253/
